pkg/interface/src/views/apps/links/components/lib/link-submit.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { Component } from 'react';
+import { hasProvider } from 'oembed-parser';
+
+import { S3Upload } from '~/views/components/s3-upload';
+import { Spinner } from '~/views/components/Spinner';
+import { Icon } from "@tlon/indigo-react";
+import GlobalApi from '~/logic/api/global';
+import { S3State } from '~/types';
+
+interface LinkSubmitProps {
+  api: GlobalApi;
+  resourcePath: string;
+  s3: S3State;
+}
+
+interface LinkSubmitState {
+  linkValue: string;
+  linkTitle: string;
+  linkValid: boolean;
+  submitFocus: boolean;
+  urlFocus: boolean;
+  disabled: boolean;
+  dragover: boolean;
+}
+
+const SubmitDragger = () => <div className="absolute bg-gray5 h-100 w-100 flex items-center justify-center z-999" style={{pointerEvents: 'none'}}>Drop a file to upload</div>
+
+export class LinkSubmit extends Component<LinkSubmitProps, LinkSubmitState> {
+  private s3Uploader: React.RefObject<S3Upload>;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      linkValue: '',
+      linkTitle: '',
+      linkValid: false,
+      submitFocus: false,
+      urlFocus: false,
+      disabled: false,
+      dragover: false
+    };
+    this.setLinkValue = this.setLinkValue.bind(this);
+    this.setLinkTitle = this.setLinkTitle.bind(this);
+    this.onDragEnter = this.onDragEnter.bind(this);
+    this.onDrop = this.onDrop.bind(this);
+    this.onPaste = this.onPaste.bind(this);
+    this.uploadFiles = this.uploadFiles.bind(this);
+    this.s3Uploader = React.createRef();
+  }
+
+  onClickPost() {
+    const link = this.state.linkValue;
+    const title = this.state.linkTitle
+      ? this.state.linkTitle
+      : this.state.linkValue;
+      this.setState({ disabled: true });
+    this.props.api.links.postLink(this.props.resourcePath, link, title).then((r) => {
+      this.setState({
+        disabled: false,
+        linkValue: '',
+        linkTitle: '',
+        linkValid: false
+      });
+    });
+  }
+
+  setLinkValid(linkValue) {
+    const URLparser = new RegExp(/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/gi);
+
+    let linkValid = URLparser.test(linkValue);
+
+    if (!linkValid) {
+      linkValid = URLparser.test(`http://${linkValue}`);
+      if (linkValid) {
+        linkValue = `http://${linkValue}`;
+      }
+    }
+
+    this.setState({ linkValid, linkValue });
+
+    if (linkValid) {
+      console.log('splarb is valid', linkValue)
```

Comment:
'splarb'?

Reply:
It's a nonsense word I've taken to using that allows you to do leakproof filtering.